tests/unit/hooks/usePhotoExif.test.js

```diff
@@ -0,0 +1,65 @@
+// Base64-encoded contents of an image file with EXIF data
+// eslint-disable-next-line max-len
+import { renderHook, waitFor } from "@testing-library/react-native";
+import { parse } from "date-fns";
+import fs from "fs";
+import { useState } from "react";
+
+import usePhotoExif, {
+  parseExifCoordinates,
+  parseExifDateTime
+} from "../../../src/components/ObsEdit/hooks/usePhotoExif";
+
+// Expected EXIF metadata of the above file
+const EXPECTED_EXIF_DATE = parse(
+  "2018-03-07 08:19:49",
+  "yyyy-MM-dd HH:mm:ss",
+  new Date()
+);
+
+const EXPECTED_EXIF_LATITUDE = 38.07480555555556;
+const EXPECTED_EXIF_LONGITUDE = -122.85112777777778;
+const EXPECTED_EXIF_POSITIONAL_ACCURACY = 5;
+
+const readFileFS = filename => (
+  fs.readFileSync(
+    filename,
+    { encoding: "base64" }
+  )
+);
+
+jest.mock( "react-native-fs", ( ) => {
+  const RNFS = {
+    // Used for testing reading EXIF data
+    read: readFileFS
+  };
+
+  return RNFS;
+} );
+
+const usePhotoExifWrapper = () => {
+  const [uri, setUri] = useState( null );
+  const exif = usePhotoExif( uri );
+
+  return { setUri, exif };
+};
+
+test( "reads and parses EXIF data from photo", async ( ) => {
+  const { result } = renderHook( () => usePhotoExifWrapper() );
+
+  await waitFor( () => {
+    // This triggers the usePhotoExif hook
+    result.current.setUri( `${process.cwd()}/tests/resources/test_exif.jpg` );
+  } );
+
+  if ( result.current.exif ) {
```

Comment:
This test only passes because this is never true.

Reply:
Ok, I think the reason it's never true is that `Exif` from `react-native-exif` isn't defined in the test env, presumably b/c it needs to have those linked native libs working and the test env is pure js. Instead of the unspecific catch you have in the hook, maybe you could make the hook have an option so you can at least test the pure JS functionality? Maybe this is all a bit premature while you're figuring out if you're going to roll your own. Not sure how people write integration tests for hooks that depend on native code.